libraries/libfc/include/fc/container/ordered_diff.hpp

```diff
@@ -26,7 +26,10 @@ namespace fc {
  * @param Container container type for ordered diff and for diff_result
  */
 template <typename T, typename SizeType = size_t, template<typename Y, typename...> typename Container = std::vector>
-requires std::equality_comparable<T> && std::random_access_iterator<typename Container<T>::iterator>
+requires std::equality_comparable<T>
+         && std::random_access_iterator<typename Container<T>::iterator>
+         && std::is_unsigned<SizeType>::value
+         && (std::numeric_limits<typename Container<T>::size_type>::max() >= std::numeric_limits<SizeType>::max())
 class ordered_diff {
 public:
    using size_type = SizeType;
```

Comment:
maybe add:
    `using cont_size_t = typename Container<T>::size_type;` 

so that you can use `cont_size_t` from now on.